programmers/비밀지도.js

```diff
@@ -0,0 +1,32 @@
+function solution(n, arr1, arr2) {
+    const passwords = [];
+    const results = [];
+    
+    let sum = 0;
+    let str = '';
+
+    for (let i = 0; i < arr1.length; i++) {
+        sum = arr1[i] | arr2[i];
+        sum = sum.toString(2);
+        passwords.push(sum)
+    }
+
+    for (let i = 0; i < passwords.length; i++) {
+        
+        for (let j = 0; j < passwords.length; j++) {
+            
+	    if (passwords[i][j] === '1') str += '#';
+            else if(passwords[i][j] === '0') str += ' ';
+        }
+
+        if (passwords[i].length !== n) {
+           
+            for (let k = 0; k < n - passwords[i].length; k++) {
+                str = ' ' + str;
+            }
+        }
+        results.push(str);
+        str = ''
+    }
```

Comment:
for문에서 i를 두번 선언해 주셨는데, 변수 선언이 중복되기 때문에 위의 있는  for문의 i 증감식이랑 연결하여 사용해 줄수 있어요!
또한 for문에서 최대값이 항상 n이기 때문에 n으로 통일해주는게 가독성이 더 좋을거 같아요 :)

```suggestion
    for (let i = 0; i < n; i++) {
        sum = (arr1[i] | arr2[i]).toString(2);
        passwords.push(sum);
    
        for (let j = 0; j < n; j++) {
	    if (passwords[i][j] === '1') str += '#';
            else if (passwords[i][j] === '0') str += ' ';
        }

        if (passwords[i].length !== n) {
            for (let k = 0; k < n - passwords[i].length; k++) {
                str = ' ' + str;
            }
        }
        results.push(str);
        str = '';
    }
```